File: frontend/web_ccadmin_lt/src/app/enterprise/sale/model/entity/CreditNoteApplicationEntity.ts
import { AuditTableEntity } from "src/app/enterprise/shared/model/entity/AuditTableEntity";

export class CreditNoteApplicationEntity extends AuditTableEntity {
    public ApplicationId: number = 0;
    public CreditNoteCod: string = "";
    public SaleCod: string = "";
    public TrxPaymentId: number = 0;
    public AmountApplied: number = 0;
}

File: backend/ws_wa_store_ccadmin/src/main/java/com/ccadmin/app/sale/model/entity/CreditNoteApplicationEntity.java
package com.ccadmin.app.sale.model.entity;

import com.ccadmin.app.shared.model.entity.AuditTableEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.io.Serializable;
import java.math.BigDecimal;

@Entity
@Table(name = "credit_note_application")
public class CreditNoteApplicationEntity extends AuditTableEntity implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long ApplicationId;
    public String CreditNoteCod;
    public String SaleCod;
    public Long TrxPaymentId;
    public BigDecimal AmountApplied;

    @Override
    public CreditNoteApplicationEntity session(String userCod) {
        this.addSession(userCod);
        return this;
    }
}
